code/game/machinery/bots/floorbot.dm

```diff
@@ -330,7 +334,7 @@ var/global/list/floorbot_targets=list()
 	// Needed because we used to look this up 15 goddamn times per process. - Nexypoo
 	var/list/shitICanSee = view(7, src)
 
-	if(src.amount <= 0 && !src.have_target())
+	if(src.amount < 50 && !src.have_target())

```

Comment:
You should add an increasing delay for searching for floor tiles.
i.e. the bot should wait a longer period of time each time before searching again to see if there are floor tiles in the vicinity, because otherwise it'll keep searching the same tiles in a costly way to no avail.